### DISCUSSION TITLE
トラックボールの動作が安定しない

### DISCUSSION THREAD

**Discussion by ut0s** [Yowkees/keyball#234] [Q&A]
トラックボールの挙動が､OLED上の値でいうと､127->0や0→127などにすぐ切り替わってしまい､マウスとして使用することが難しい状態です｡
トラックボール基盤･ジャンパの再はんだ､等は実施した状態になります｡
左右両方のPromicroに接続して試してみたのですが､同じ結果でCPIを変化させても解決しませんでした｡

はんだが怪しそうなのは他のDiscussionを見て感じているのですが､自己解決できませんでした｡
解決策ご教授お願いいたします｡

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
状況了解しました。

ミドル基板かトラックボール基板の半田付け不良が原因と思われます。
はんだ付け面のお写真を見せていただくことは可能でしょうか？

また、白銀ラボのお問合せからメッセージをいただければそちらから回答も可能ですのでご検討ください。
https://shirogane-lab.com/pages/contact

**Comment by ut0s** [comment]
回答遅くなりました｡

色々ためしたのですが､自力解決できませんでした...
写真の添付させていただきます

|マイコン1 |マイコン2| 7Pコンスルー | センサー表面|
|-- | -- | --|--|
| ![IMG_0087](https://github.com/Yowkees/keyball/assets/44089530/f25b32e9-7f35-427c-a57b-dc392f8a0ba4) |![IMG_0088](https://github.com/Yowkees/keyball/assets/44089530/3fc65111-a32d-4993-b4f9-3adc3f7437e1)  |![IMG_0086](https://github.com/Yowkees/keyball/assets/44089530/41c6ce62-4888-4a13-9e60-28ad4078b639)  | ![IMG_0085](https://github.com/Yowkees/keyball/assets/44089530/7a9eed0a-db7b-42e0-96e5-7043a4df0f97) |

下記に従いマイコンと7Pの通電も確認しましたが､問題ありませんでした｡
https://github.com/Yowkees/keyball/discussions/276#discussioncomment-6483429

**Comment by Yowkees** [comment]
お写真ありがとうございます。
半田付けはキレイにできていると思います。

１点、L字7Pコンスルーの接触不良を確認したいと思います。
コンスルーが斜めに取り付けられている、ピンに変形がある場合、動作不良となった事例がありますので、ご確認ください。
また、トラックボール基板は奥まで差し込んでください。

次にProMicroが原因の可能性があります。
写真を見ますと、白銀ラボで取り扱っているProMicroではなさそうですがどこで入手されたものでしょうか？
自作キーボードショップで多く流通しているProMicroに交換したところ改善したという件が過去にありましたので、最後の手段で交換もご検討いただければと思います。

以上、お手数ですがご検討お願いいたします。

**Reply by ut0s** [reply]
L字7Pコンスルーも問題なさそうでしたので､ProMicroの交換を進めたいと思います
ありがとうございました

→ ProMicro交換後無事に動作するようになりました｡ありがとうございます

**Reply by risn2** [reply]
当方も同様の症状になり、こちらのアドバイスを参考に、
amazonで購入したProMicroから白銀ラボで
購入したものに交換したところ、
症状が改善しました。
amazonの商品はキースイッチに関する問題はありませんでしたが、
トラックボール基板で使用するSPI通信に問題があるようです。

ご参考まで。
